File: acmicpc/18234.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

type Data struct {
	add int
	base int
}

var bufin = bufio.NewReader(os.Stdin)
var bufout = bufio.NewWriter(os.Stdin)
var n, t, x, y int
var vec []Data

func main() {
	defer bufout.Flush()
	fmt.Fscan(bufin, &n, &t);

	for i:=0; i<n; i++ {
		fmt.Fscan(bufin, &x, &y);
		vec = append(vec, Data{y, x});
	}

	sort.Slice(vec, func(i, j int) bool {
		if vec[i].add == vec[j].add {
			return vec[i].base < vec[j].base
		} else {
			return vec[i].add < vec[j].add
		}
	});

	var ans int64
	for i:=0; i<n; i++ {
		ans += int64(vec[i].base) + int64(vec[i].add) * int64(t-n+i)
	}

	fmt.Println(ans)
}

File: acmicpc/18234.cpp
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;

int n, t, x, y;
vector<pair<int, int>> vec;

int main()
{
	ios_base::sync_with_stdio(0);
	cout.tie(0);
	cin.tie(0);

	cin >> n >> t;
	for (int i=0; i<n; ++i) {
		cin >> x >> y;
		vec.push_back({y, x});
	}

	sort(vec.begin(), vec.end());

	long long ans = 0;
	for (int i=0; i<n; ++i) {
		auto [add, base] = vec[i];
		ans += base + 1LL * add * (t-n+i);
	}
	cout << ans << '\n';
	
	return 0;
}
